### ISSUE TITLE
Access current span in NodeJS instrumentation

### ISSUE THREAD

**Issue by santoshkashyap** [open-telemetry/opentelemetry-operator#1331]
I have OpenTelemetry Operator configured our my Kubernetes cluster. For NodeJS application the instrumentation works fine by using the annotation:
```
# format: <namespace>/<Instrumentation CR name>
instrumentation.opentelemetry.io/inject-nodejs=dev-opentelemetry/opentelemetry-instrumentation
```

Auto-instrumentation is working fine and the traces and spans are generated as expected. However, I have a scenario where I need access to active span and propagate it as W3C traceparent manually.

Below are a couple of TypeScript snippets using `@opentelemetry/api` that I tried in vain (based on the documentation: [get-the-current-span](https://opentelemetry.io/docs/instrumentation/js/instrumentation/#get-the-current-span)) :

```
import { trace } from '@opentelemetry/api'
.....
const activeSpan = trace.getActiveSpan()

console.log('activeSpan', activeSpan)  // active span value is "undefined"

const traceId = activeSpan?.spanContext().traceId
const spanId = activeSpan?.spanContext().spanId
const traceFlag = activeSpan?.spanContext().traceFlags

Also tried below API:

const currentSpan = trace.getSpan(context.active()).

console.log(currentSpan) // Again currentSpan is "undefined"

```

Did I miss additional configuration or setup for OpenTelemetry access ? It would be nice to have access to current span programmatically when using Operator based instrumentation. Any inputs appreciated. Thanks in advance!!!

**Best,**
Santosh

**Comment by pavolloffay** [comment]
I would suggest instrumenting your project explicitly without the operator and seeing if the current span is present. 

The operator just installs the nodejs libraries, it does not use any special instrumentation. A better place to open this issue would be the nodejs repository.

**Comment by santoshkashyap** [comment]
> I would suggest instrumenting your project explicitly without the operator and seeing if the current span is present.

Yes, explicit instrumentation works without any issues (example defining my `tracing.js` and running it with `NODE_OPTIONS= --require ./src/tracing.js`).  This way, I was also able to access the current span. The problem comes with the operator injected approach.  I was under the impression that opentelemetry api should be able to access the global trace injected by the operator.

**Comment by pavolloffay** [comment]
All operator injection code is listed here, so you should be able to replicate it locally https://github.com/open-telemetry/opentelemetry-operator/tree/main/autoinstrumentation/nodejs

**Comment by mat-rumian** [comment]
Hey @santoshkashyap,
I think it's very important in which place in your code you want to get `spanContext`. I was able to get a context and log all the span details using:

```
const { trace, context }  = require("@opentelemetry/api");

let activeSpan = trace.getSpan(context.active());

const traceId = activeSpan.spanContext().traceId
const spanId = activeSpan.spanContext().spanId
const traceFlag = activeSpan.spanContext().traceFlags

console.log(traceId, spanId, traceFlag)
```

but I needed to find a place where span is not undefined.

**Comment by santoshkashyap** [comment]
Thank you for your response @pavolloffay, @mat-rumian.

I figured the issue seems to be with the version compatibility between `@opentelemetry/api` and `autoinstrumentation-nodejs` injected by the OpenTelemetry Operator . I was using the latest version  for `@opentelemetry/api@1.3.0`.  However, the `autoinstrumentation-nodejs` injected by the operator had the version `0.31.0` (verified it via the init container's `image: ghcr.io/open-telemetry/opentelemetry-operator/autoinstrumentation-nodejs:0.31.0`). With this setup, I receive "Undefined" values for **activeSpan** and consequently also for **traceId** and **SpanId**.

Based on the API in the above sample, and also referring to commit history https://github.com/open-telemetry/opentelemetry-operator/commit/e341f288d2000ebb030b06654f5763c1ec3855f8
I tried out `@opentelemetry/api` version `1.1.0`. Now I'm able to access the `traceId` and `spanId`.  I assume, this also implies I need to take care of updating the API version in future to avoid any breaking changes ?

I'm fine to close the ticket (If you don't have further remarks). Thank you
